### PR TITLE
Fix failing tests for locators module

### PR DESCRIPTION
**Pull request by pedro-w**
Fixed tests that were failing in the `locators-module-test-suite` 
Also, updated the library documentation for this module.

### CODE REVIEW

File: sources/system/tests/locators.dylan [hunk 1]
```diff
@@ -115,100 +130,237 @@ define locators class-test <https-server> ()
...
 
-define locators class-test <cgi-url> ()
-  //---*** Fill this in...
+ define locators class-test <cgi-url> ()
+  let server = make(<http-server>,
+                    host: "www.functionalobjects.com",
+                    protocol: "http");
+  let directory = make(<directory-url>,
+                       server: server,
+                       path: #["path","to","resource"]);
```
Comment (cgay) [inline sources/system/tests/locators.dylan]: Always a space after comma, unless at end of line.

File: sources/system/tests/locators.dylan [hunk 2]
```diff
@@ -115,100 +130,237 @@ define locators class-test <https-server> ()
...
+  assert-equal("example", locator-name(as(<file-locator>, "example")));
+  assert-equal("example.dylan", locator-name(as(<file-locator>, "/home/andrewa/example.dylan")));
 end function-test locator-name;
 
 /// Coercion protocols
 
 define locators function-test locator-as-string ()
-  //---*** Fill this in...
+  let server = make(<http-server>,
+                    host: "www.functionalobjects.com",
```
Comment (cgay) [inline sources/system/tests/locators.dylan]: A shorter host (foo.com) would shorten some assertion lines below.

**Comment by pedro-w** [comment]
Thanks for the comments @cgay, I will update the PR accordingly.

File: sources/system/locators/web-locators.dylan [hunk 3]
```diff
@@ -484,6 +507,19 @@ define sealed method locator-as-string
...
 
+define method locator-path
+  (locator :: <cgi-url>) => (path :: <sequence>)
+  locator-path(locator-file(locator))
+end method locator-path;
+
+define sealed method \=
+  (locator1 :: <cgi-url>,
+   locator2 :: <cgi-url>)
+  => (equal? :: <boolean>)
```
[the remark this answers is not in the document]
Comment (pedro-w) [reply sources/system/locators/web-locators.dylan]: It certainly could. I copied that layout from elsewhere in the file, e.g. https://github.com/dylan-lang/opendylan/blob/5e636b9257572ce1177c881842d895332d3b017a/sources/system/locators/web-locators.dylan#L327-L331
I will make them all consistent.

**Pull request merged**